### DISCUSSION TITLE
Questions around building on Mac OS

### DISCUSSION THREAD

**Discussion by bmustill-rose** [pyinstaller/pyinstaller#7417] [Help] [answered]
Hi all,

I've developed a Windows app using PyInstaller that's become (somewhat) popular and am currently talking to a charity who are looking to fund the purchase of a Mac so I can develop a Mac version. I'm comfortable with iOS development through work but don't have huge amounts of Mac OS experience outside of this.

My questions (apologies if some are considered OT):

- I'd like my app to work on Intel and Apple silicon. From reading my current plan for doing this is to source an AS-based machine then build the app in a virtualized copy of Catalina (looking for broad OS compatibility) using something like UTM. Is this a sensible approach? Any PyInstaller specifics I should be aware of? (Appreciate this isn't a UTM discussion forum).
- Code signing: it looks like this https://github.com/txoof/codesign repo / the steps in it are the way to go. What are peoples actual experiences around this process? I'm particularly interested in how well or otherwise notorization works for PyInstaller apps? Or is PyInstaller not relevant here?
- Does anyone have any thoughts around how viable app store submission is for PyInstaller-based apps?

Many thanks.

**Comment by bwoodsend** [comment, marked as answer]
> I'd like my app to work on Intel and Apple silicon. From reading my current plan for doing this is to source an AS-based machine then build the app in a virtualized copy of Catalina (looking for broad OS compatibility) using something like UTM. Is this a sensible approach? Any PyInstaller specifics I should be aware of? (Appreciate this isn't a UTM discussion forum).

Without going down the misery path of `universal2` (combined architectures in one build), you can build for `x86_64` and `arm64` on an M1 machine but you can only build for `x86_64` on an Intel machine. However, you can't run Catolina on `arm64` (UTM won't help you across architectures) so you can't cover both paths with a single machine. CI/CD is the way to solve this though - [Cirrus CI](https://cirrus-ci.org/) is currently the only platform supporting macOS `arm64`.

> Code signing: it looks like this https://github.com/txoof/codesign repo / the steps in it are the way to go. What are peoples actual experiences around this process? I'm particularly interested in how well or otherwise notorization works for PyInstaller apps? Or is PyInstaller not relevant here?

It's a miserable process. 

> Does anyone have any thoughts around how viable app store submission is for PyInstaller-based apps?

Also miserable but it's been done once. See #7123.

**Reply by bmustill-rose** [reply]
Right, that's all good to know. Thank you. Looks like this is shaping up to be quite the adventure. I'll surface my findings somewhere appropriate if I encounter anything interesting.
